Compile issue with libdns v0.2.2

The module doesn't compile with libdns v0.2.2:

`/go/pkg/mod/github.com/libdns/infomaniak@v0.1.2/mappers.go:23:13: cannot use int(ikr.Priority) (value of type int) as uint value in struct literal`

Therefore, the [caddy-dns/infomaniak](https://github.com/caddy-dns/infomaniak) cannot be used with caddy v2.8 ([see this issue](https://github.com/caddy-dns/infomaniak/issues/2))

--- a/go.mod
+++ b/go.mod
@@ -2,4 +2,4 @@ module github.com/libdns/infomaniak
 
 go 1.18
 
-require github.com/libdns/libdns v0.2.1
+require github.com/libdns/libdns v0.2.2

--- a/mappers.go
+++ b/mappers.go
@@ -20,7 +20,7 @@ func (ikr *IkRecord) ToLibDnsRecord(zone string) libdns.Record {
 		Name:     libdns.RelativeName(ikr.SourceIdn, zone),
 		Value:    ikr.Target,
 		TTL:      time.Duration(ikr.TtlInSec),
-		Priority: int(ikr.Priority),
+		Priority: ikr.Priority,
 	}
 }
 
@@ -32,7 +32,7 @@ func ToInfomaniakRecord(libdnsRec *libdns.Record, zone string) IkRecord {
 		SourceIdn: libdns.AbsoluteName(libdnsRec.Name, zone),
 		Target:    libdnsRec.Value,
 		TtlInSec:  uint(libdnsRec.TTL),
-		Priority:  uint(libdnsRec.Priority),
+		Priority:  libdnsRec.Priority,
 	}
 
 	if ikRec.TtlInSec <= 0 {

--- a/mappers_test.go
+++ b/mappers_test.go
@@ -20,7 +20,7 @@ func Test_ToLibDnsRecord_MapsAllProperties(t *testing.T) {
 	assertEquals(t, "Type", ikRec.Type, libRec.Type)
 	assertEquals(t, "Value", ikRec.Target, libRec.Value)
 	assertEqualsInt(t, "TTL", int(ikRec.TtlInSec), int(int64(libRec.TTL)))
-	assertEqualsInt(t, "Priority", int(ikRec.Priority), libRec.Priority)
+	assertEqualsInt(t, "Priority", int(ikRec.Priority), int(libRec.Priority))
 }
 
 func Test_ToLibDnsRecord_ReturnsRelativeName(t *testing.T) {
@@ -49,7 +49,7 @@ func Test_ToInfomaniakRecord_MapsAllProperties(t *testing.T) {
 	assertEquals(t, "Type", libRec.Type, ikRec.Type)
 	assertEquals(t, "Value", libRec.Value, ikRec.Target)
 	assertEqualsInt(t, "TTL", int(libRec.TTL), int(ikRec.TtlInSec))
-	assertEqualsInt(t, "Priority", libRec.Priority, int(ikRec.Priority))
+	assertEqualsInt(t, "Priority", int(libRec.Priority), int(ikRec.Priority))
 }
 
 func Test_ToInfomaniakRecord_DefaultTtlIsAppliedIfNoTtlProvided(t *testing.T) {

--- a/provider_test.go
+++ b/provider_test.go
@@ -64,7 +64,7 @@ func Test_GetRecords_ReturnsRecords(t *testing.T) {
 	assertEqualsInt(t, "TTL", int(expectedRec.TtlInSec), int(actualRec.TTL))
 	assertEquals(t, "Type", expectedRec.Type, actualRec.Type)
 	assertEquals(t, "Value", expectedRec.Target, actualRec.Value)
-	assertEqualsInt(t, "Priority", int(expectedRec.Priority), actualRec.Priority)
+	assertEqualsInt(t, "Priority", int(expectedRec.Priority), int(actualRec.Priority))
 }
 
 func Test_GetRecords_RemovesTrailingDotsFromZone(t *testing.T) {

